### ISSUE TITLE
New toolbar for execution of commands and access to preview

### ISSUE THREAD

**Issue by vparfonov** [eclipse-che/che#3519]
See the toolbar integrated in the IDE in the following mockup:

![servers-ports as editor copy](https://cloud.githubusercontent.com/assets/1636769/22382354/a0361900-e4c5-11e6-8d47-8f5392a61a42.png)

There are multiple states which needs to be handled:

## No action/commands in progress
<img width="899" alt="screen shot 2017-01-24 at 22 17 42" src="https://cloud.githubusercontent.com/assets/1636769/22267140/f2e99ae0-e282-11e6-9cd0-46d450a7d971.png">

### General view
![screen shot 2017-01-27 at 19 20 54](https://cloud.githubusercontent.com/assets/1636769/22382383/bffb1ef2-e4c5-11e6-9b6e-d985701911af.png)

### Details
![screen shot 2017-01-27 at 19 21 18](https://cloud.githubusercontent.com/assets/1636769/22382398/cf7d1c0e-e4c5-11e6-8143-d68afa401b6f.png)

### Click on the command
Clicking on the command, will show the console output associated for the command in Console panel.

### Hover on command
![screen shot 2017-01-27 at 19 21 49](https://cloud.githubusercontent.com/assets/1636769/22382412/e24e4c4a-e4c5-11e6-909e-4cc29170ce9c.png)

We display the whole command in the hover and adapt the width properly so it displays nicely with word-wrappind

### Preview
![screen shot 2017-01-27 at 19 22 14](https://cloud.githubusercontent.com/assets/1636769/22382431/f3d90608-e4c5-11e6-9ef4-6e58e2581221.png)

When the user clicks on "Preview" it displays a small dropdown with all the previews for all currently running commands (started by the user, not the ones from the agents).

<img width="206" alt="screen shot 2017-01-24 at 22 32 39" src="https://cloud.githubusercontent.com/assets/1636769/22267673/048201f0-e285-11e6-8b39-6dbc8481fad7.png">

They are ordered by the last one added into the list.
Example:
- the user call a command1 with PreviewURL1
- then the user call a command2 with PreviewURL2 
User will see:
- PreviewURL2
- PreviewURL1

When the user clicks on a previewURL, the URL is opened on a separated browser's tab.

If the command that is providing a previewURL is stopped, then the previewURL gets removed from the list displayed in the dropdown.

**Note:** Later user should be able to pin a preview url, to default the one opened with a single click on "Preview".

User can click on it to open the preview in another browser’s tab.
## Command Finished
![screen shot 2017-01-27 at 19 22 54](https://cloud.githubusercontent.com/assets/1636769/22382448/089d6476-e4c6-11e6-959c-8e0f47287fe9.png)

The button allows to relaunch the command.

## Expanded activity monitor (Phase 2)
![screen shot 2017-01-27 at 19 23 37](https://cloud.githubusercontent.com/assets/1636769/22382472/200db444-e4c6-11e6-83b2-92d9894da85a.png)

Commands displayed in the list:
- the list is displaying all commands executed accross all machines
- a maximum of 7 commands is shown in the dropdown, but the dropdown can be scrolled

The commands displayed are sync with the commands displayed in the Console panel
- if a command is removed from the command panel, then it is cleared from the Activity Monitor

There is no icon showing that a command is running. 
- This information is visible for the user, by seeing the duration updated each seconds

Multiple commands can be running at the same time.
- When a new command is invoked, it takes precedence and replace the previous one executed in the header.

If user selects a command in the list, it displays it in the header and show the corresponding command outputs in the Console Panel.

## Run and Debug buttons
![screen shot 2017-01-24 at 22 36 41](https://cloud.githubusercontent.com/assets/1636769/22267856/9ab0b360-e285-11e6-9fdc-9c59415c4f19.png)

By default, we display buttons for “Run” and “Debug”. The buttons when clicked, it displays the whole commands list of the goal and when a command is hovered, it allows to select the target where to execute the command.

**Nice to have:** 
User must have the ability to define default command for a goal.
I’d like the user being able to configure a button per goal in the toolbar if he wants it. This should be in the Preferences, but by default, we would display “Run” and “Build” buttons in the toolbar.
We would have two actions on these buttons:
- single-click: it executes the default command of the goal
- long-click: it will display the whole commands list of the goal in a dropdown as we have today.

Linked to: #2681

**Comment by vparfonov** [comment]
We have goals. They need to be used for simplifying the user experience.
By default, we will show button for “Run” and “Debug”, but I’d like the user being able to configure a button per goal in the toolbar if he wants it. This should be in the Preferences, but by default, we would display “Run” and “Build” buttons in the toolbar. 
We would have two actions on these buttons:
- single-click: it executes the default command of the goal
- long-click: it will display the whole commands list of the goal in a dropdown as we have today. 
User must have the ability to define default command for a goal.

**Comment by slemeur** [comment]
Updated specification

**Comment by slemeur** [comment]
@vparfonov + @azatsarynnyy : I have updated the requirements.

Base on the discussion we had with @azatsarynnyy this morning, we will move forward without implementing the "dropdown menu" when clicking on the controller. This gives you the ability to continue your work without being block on this part of the component.
In parallel, we can continue to discuss the specification and adjust it depending on technical challenges. 

Thanks
